project-base/storefront/store/useCookiesStore.ts

```diff
@@ -1,27 +1,47 @@
 import { CookiesStoreContext } from 'components/providers/CookiesStoreProvider';
 import { useContext } from 'react';
+import { v4 as uuidV4 } from 'uuid';
 import { useStore } from 'zustand';
 import { createStore } from 'zustand/vanilla';
 
-export type CookiesStoreState = { lastVisitedProductsCatnums: string[] | undefined };
+export type CookiesStoreState = {
+    lastVisitedProductsCatnums: string[] | null;
+    userIdentifier: string;
+};
 
 export type CookiesStoreActions = {
     setCookiesStoreState: (value: Partial<CookiesStoreState>) => void;
 };
 
 export type CookiesStore = CookiesStoreState & CookiesStoreActions;
 
-export const defaultInitState: CookiesStoreState = {
-    lastVisitedProductsCatnums: undefined,
-};
+export const getDefaultInitState = (): CookiesStoreState => ({
+    lastVisitedProductsCatnums: null,
+    userIdentifier: uuidV4(),
+});
 
-export const createCookiesStore = (initState: CookiesStoreState = defaultInitState) =>
+export const createCookiesStore = (cookieStoreFromServer: CookiesStoreState) =>
     createStore<CookiesStore>()((set) => ({
+        ...cookieStoreFromServer,
+        setCookiesStoreState: (value) => {
+            set((state) => ({ ...state, ...value }));
+        },
+    }));
+
+export const createCookiesStoreOnServer = (cookieStoreStateFromContext: string | null) => {
+    let initState = getDefaultInitState();
+
+    if (cookieStoreStateFromContext) {
+        initState = JSON.parse(cookieStoreStateFromContext);
+    }
+
+    return createStore<CookiesStore>()((set) => ({
         ...initState,
         setCookiesStoreState: (value) => {
```

Comment:
I actually renamed it to `setCookiesStoreStateOnClient` to make sure everybody knows it can only be used on the client and removed it from the server version, as it was not used. This also resolved the issue with serialization